Sources/Core/Sources/ECS/Systems/PlayerInputSystem.swift

```diff
@@ -146,6 +148,47 @@ public final class PlayerInputSystem: System {
         if !message.isEmpty {
           guiState.messageInput?.removeLast()
         }
+      } else if event.key == .upArrow {
+        // If no message is selected, select the above message
+        if guiState.currentMessageIndex == nil {
+          guiState.stashedMessageInput = guiState.messageInput
+          // If there are no messages, do nothing
+          if guiState.playerMessageHistory.count > 0 {
+            // Index up a message
+            guiState.currentMessageIndex = guiState.playerMessageHistory.count - 1
+            if guiState.currentMessageIndex == guiState.currentMessageIndex {
```

Comment:
Am I missing something here, or…?

Reply:
it’s to make sure the value isn’t nil without force unwrapping

Reply:
Surely that should be 
`if let currentMessageIndex = guiState.currentMessageIndex { }`
unless my swift is rusty?

Reply:
Because `if nil == nil { }` is surely true

Reply:
yeah sorry, still getting the hang of swift